internal/querynode/segment_loader.go

```diff
@@ -623,7 +623,7 @@ func (loader *segmentLoader) checkSegmentSize(collectionID UniqueID, segmentLoad
 	}
 
 	// when load segment, data will be copied from go memory to c++ memory
-	if usedMemAfterLoad+maxSegmentSize > int64(float64(totalMem)*Params.QueryNodeCfg.OverloadedMemoryThresholdPercentage) {
+	if uint64(usedMemAfterLoad+maxSegmentSize) > uint64(float64(totalMem)*Params.QueryNodeCfg.OverloadedMemoryThresholdPercentage) {
```

Comment:
It would be better to change the types of usedMemAfterLoad and maxSegmentSize to uint64.

Reply:
Why not convert to float64？

Reply:
low-level calls hardware_info.GetMemoryCount method return uint64,

Reply:
> Why not convert to float64？

low-level calls hardware_info.GetMemoryCount method return uint64, use 'cgroups' and 'gopsutil' return numeric type is uint64，I think max support is uint64，no need to support more memory.  This is just to fix the problem of large memory overflow. now the error message is "OOM", not quite right.